src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageUtility.java

```diff
@@ -16,15 +21,25 @@
  */
 
 public interface IridaFileStorageUtility {
-	//Valid extensions to try to concatenate with this tool
-	public static final List<String> VALID_EXTENSIONS = Lists.newArrayList("fastq", "fastq.gz");
+	Logger logger = LoggerFactory.getLogger(IridaFileStorageUtility.class);
+
+	//Valid file extensions for sample file concatenation
+	public static final List<String> VALID_CONCATENATION_EXTENSIONS = Lists.newArrayList("fastq", "fastq.gz");
 	/**
 	 * Get a temporarry file from storage
 	 *
 	 * @param file The {@link Path} to the file
-	 * @return {@link File} which was retrieved from path
+	 * @return {@link IridaTemporaryFile} which includes the file and optional temporary directory
+	 */
+	public IridaTemporaryFile getTemporaryFile(Path file);
+
+	/**
+	 * Delete temporary downloaded file and/or directory.
+	 *
+	 * @param filePath The {@link Path} to the file
+	 * @param directoryPath The {@link Path} to the directory which has the file
 	 */
-	public File getTemporaryFile(Path file);
+	public void cleanupDownloadedLocalTemporaryFiles(Path filePath, Path directoryPath);
```

Comment:
For this method please accept an `IridaTemporaryFile` instead of the 2 `Path`s.  That will help ensure we only use this in the appropriate cases.

Reply:
Updated in 1dbdb47af3eedeee57410916380b562e92a85c41